src/vault.rs

```diff
@@ -58,6 +60,21 @@ impl VaultBuilder {
                                         continue;
                                     }
                                 }
+
+                                if self.whitelist_mode {
+                                    let publish = note.metadata.get("publish");
+                                    match publish {
+                                        Some(publish) => {
+                                            if let MetadataValue::Boolean(publish) = publish {
+                                                if !publish {
+                                                    continue;
+                                                }
+                                            }
+                                        }
+                                        None => continue,
+                                    }
+                                }
+
```

Comment:
I think that a `as_bool` helper fn on MetadataValue can be useful here:

```suggestion

                                                                if self.whitelist_mode {
                                    let publish = note
                                        .metadata
                                        .get("publish")
                                        .map_or(Some(false), |value| value.as_bool())
                                        .unwrap_or(false);
                                    if !publish {
                                        continue;
                                    }
                                }

```

```rust
impl MetadataValue {
    pub fn as_bool(&self) -> Option<bool> {
        match self {
            MetadataValue::Boolean(value) => Some(*value),
            _ => None,
        }
    }
}

```